cocos/navmesh/CCNavMesh.cpp

```diff
@@ -32,20 +32,22 @@
 
 NS_CC_BEGIN
 
+#pragma pack(push,1)
 struct TileCacheSetHeader
 {
-    int magic;
-    int version;
-    int numTiles;
+    int32_t magic;
+    int32_t version;
+    int32_t numTiles;
```

Comment:
`TileCacheSetHeader` is used to read the navMesh file. So its size must be exact.
two things I fixed:

* added `pragma pack(1)`
* replaced `int` with `int32_t`, since the size of `int` varies depending on the architeture